apps/server/src/api-data/rundown/rundown.service.ts

```diff
@@ -9,21 +11,21 @@ import {
   PatchWithId,
   Rundown,
 } from 'ontime-types';
+import { customFieldLabelToKey } from 'ontime-utils';
 
-import { getPreviousId } from '../../services/rundown-service/rundownUtils.js';
 import { updateRundownData } from '../../stores/runtimeState.js';
 import { sendRefetch } from '../../adapters/websocketAux.js';
 import { runtimeService } from '../../services/runtime-service/RuntimeService.js';
 
-import { createTransaction, rundownCache, rundownMutation } from './rundown.dao.js';
-import { RundownMetadata } from './rundown.types.js';
-import { generateEvent, hasChanges } from './rundown.utils.js';
+import { createTransaction, customFieldMutation, rundownCache, rundownMutation } from './rundown.dao.js';
+import type { RundownMetadata } from './rundown.types.js';
+import { generateEvent, getInsertAfterId, hasChanges } from './rundown.utils.js';
 
 /**
  * creates a new entry with given data
  */
 export async function addEntry(eventData: EventPostPayload): Promise<OntimeEntry> {
-  const { rundown, commit } = createTransaction();
+  const { rundown, commit } = createTransaction(true, false);
```

Comment:
would it maybe be easier to understand if it was just two functions
`createRundownTransaction`  and `createCustomFieldTransaction` 
that just wraps this functionality